### PR TITLE
Cache empty class search

### PR DESCRIPTION
**Pull request by rlepinski**
Cache empty class results when searching a package. This prevents a class from searching the same packages over and over again if it appears multiple times in the document.

My local docs project build time went from ~4m30s to ~2m45s.

### CODE REVIEW

**Comment by bluekeyes** [comment]
Thanks for the PR. It's hard to argue with that speed increase and I can't think of anything this would break, but I also can't figure out how it improved the performance so dramatically in a normal document. The main case I found was if you have many references to a (deeply) nested class (e.g. `com.example.Server$Configuration.Connection`), but that seems uncommon.

Is this mostly to improve the case where you have an unknown reference or leave something off the classpath? Or does this also improve performance in documents that otherwise have completely valid references?

File: javalink/loader.py [hunk 1]
```diff
@@ -66,9 +66,8 @@ def load(self, name):
                     msg = "Wanted class '{}', but '{}' was loaded"
                     raise ValueError(msg.format(name, clazz))
 
-                classes = self.packages.setdefault(package, {})
-                classes[class_name] = clazz
```
Comment (bluekeyes) [inline javalink/loader.py:70]: Now that this happens in all cases, please combine the two `if` statements on L64 and L65:

``` python
if clazz and (clazz.package != package or clazz.name != class_name):
```

**Comment by rlepinski** [comment]
The time change listed is worse case scenario usage of the plugin. We reference 2 large jars (android, google play services) and 1 small jar (urban airship). All references are just class names and we have several imports. However only a few embedded classes (RichPushManager.Listener, etc...).

For each import, the plugin was searching each jar for that import every time the class was referenced. Switching to fully qualified class names reduces the build time to around 30 seconds with or without this change.

I was actually able to bring the initial build time down to 30 seconds and sequential build down to 7 seconds by pre-populating the packages with every class from the jars on init. I am working on a PR for that change as well.

**Comment by bluekeyes** [comment]
Oh, right, searching all imports - missed that when I was looking last night. This looks good now, but I'll hold off on making a new release to wait for your other optimization.

**Comment by bluekeyes** [comment]
Now available in 0.11.0 on PyPI

**Pull request merged**